app/models.py

```diff
@@ -129,12 +129,25 @@ def validates_framework(self, key, framework):
 
     slug_pattern = re.compile("^[\w-]+$")
 
-    @validates("slug")
+    @validates('slug')
     def validates_slug(self, key, slug):
         if not self.slug_pattern.match(slug):
             raise ValidationError("Invalid slug value '{}'".format(slug))
         return slug
 
+    @validates('clarification_questions_open')
+    def validates_clarification_questions_open(self, key, value):
+        # Note that because of a race condition between this key and `status`, it _is_ possible to be in a state where
+        # `clarification_questions_open` is `True` and the new status is not one of the allowed ones if `self.status`
+        # *at the time of the update* is one of the allowed ones
```

Comment:
Well spotted. Is there anything we can/want to do about this or are we happy to just be aware of it?